nucliadb_telemetry/nucliadb_telemetry/logs.py

```diff
@@ -0,0 +1,192 @@
+            data[output_property] = getattr(record, property, None)
+
+        data["logging.googleapis.com/sourceLocation"] = {
+            "file": record.pathname,
+            "line": record.lineno,
+            "function": record.funcName,
+        }
+
+        current_span = trace.get_current_span()
+        if current_span not in (INVALID_SPAN, None):
+            span_context = current_span.get_span_context()
+            # for us, this is opentelemetry trace_id/span_id
+            # GCP has logging.googleapis.com/spanId but it's for it's own cloud tracing system
+            data["trace_id"] = span_context.trace_id
+            data["span_id"] = span_context.span_id
+
+        if hasattr(record, "stack_info"):
+            data["stack_info"] = record.stack_info
+        else:  # pragma: no cover
+            data["stack_info"] = None
+
+    def extra_from_record(self, record):
+        return {
+            attr_name: record.__dict__[attr_name]
+            for attr_name in record.__dict__
+            if attr_name not in _BUILTIN_ATTRS
+        }
+
+
+class UvicornAccessFormatter(logging.Formatter):
+    def format(self, record: logging.LogRecord) -> str:
+        recordcopy = copy(record)
+        (
+            client_addr,
+            method,
+            full_path,
+            http_version,
+            status_code,
+        ) = recordcopy.args  # type: ignore[misc]
+        request_line = "%s %s HTTP/%s" % (method, full_path, http_version)
+        recordcopy.__dict__.update(
+            {
+                "httpRequest": {
+                    "requestMethod": method,
+                    "requestUrl": full_path,
+                    "status": status_code,
+                    "remoteIp": client_addr,
+                    "protocol": http_version,
+                },
+                "message": request_line,
+            }
+        )
+        return super().format(recordcopy)
+
+
+_ACCESS_LOGGER_NAME = "uvicorn.access"
+
+
+def setup_logging() -> None:
+    settings = LogSettings()
+
+    if settings.logger_levels is None:
+        settings.logger_levels = {}
+    if _ACCESS_LOGGER_NAME not in settings.logger_levels:
+        settings.logger_levels[_ACCESS_LOGGER_NAME] = settings.log_level
+
+    formatter = JSONFormatter()
+    access_formatter = UvicornAccessFormatter()
+
+    json_handler = logging.StreamHandler()
+    json_handler.setFormatter(formatter)
+    access_handler = logging.StreamHandler()
+    access_handler.setFormatter(access_formatter)
+    stream_handler = logging.StreamHandler()
+
+    root_logger = logging.getLogger()
+    access_logger = logging.getLogger(_ACCESS_LOGGER_NAME)
+    if not settings.dev:
+        root_logger.addHandler(json_handler)
+        access_logger.handlers = []
+        access_logger.addHandler(access_handler)
+    else:
+        root_logger.addHandler(stream_handler)
+
+    root_logger.setLevel(getattr(logging, settings.log_level.value))
```

Comment:
Settings `LogLevel`s are coupled to logging log level attributes and it's not explicit. A change on the enum names can break this code.